app/server/ruby/lib/sonicpi/lang/core.rb

```diff
@@ -3418,6 +3418,103 @@ def with_random_seed(seed, &block)
 
 
 
+      def use_random_type(noise_type, &block)
+        raise ArgumentError, "use_random_type does not work with a block. Perhaps you meant with_random_type" if block
+        raise ArgumentError, "invalid noise type '#{noise_type}' - please use one of :white, :pink or :perlin instead" unless %w(white pink perlin).include?(noise_type.to_s)
```

Comment:
If you need these to be symbols, I think you can just do this:
```suggestion
        raise ArgumentError, "invalid noise type '#{noise_type}' - please use one of :white, :pink or :perlin instead" unless %i[white pink perlin].include?(noise_type)
```